eitprocessing/eit_data/__init__.py

```diff
@@ -208,23 +207,18 @@ def concatenate(cls, a: T, b: T, label: str | None = None) -> T:
             variants=variants,
         )
 
-    @classmethod
-    def check_equivalence(cls, a: T, b: T, raise_=False) -> bool:
-        cm = contextlib.nullcontext() if raise_ else contextlib.suppress(NotEquivalent)
-        with cm:
-            if a.__class__ != b.__class__:
-                raise NotEquivalent(f"Classes don't match: {type(a)}, {type(b)}")
-
-            if a.framerate != b.framerate:
-                raise NotEquivalent(
-                    f"Framerates do not match: {a.framerate}, {b.framerate}"
-                )
-
-            VariantCollection.check_equivalence(a.variants, b.variants, raise_=True)
-
-            return True
-
-        return False
+    def isequivalent(
+        self,
+        other: Self,
```

Comment:
I understand your argumentation here. However, because type hints are not enforced in python, you can still pass anything to `foo` and `bar` without a problem. The parent method will check whether they have the same type and if not return `False`.
The reason I wanted I wrote it as `other: Self` is that now pylance will recognize that it is expected to be the same type as `self` and therefore help while coding in the IDE (color coding, referencing arguments, etc). 
Also note that no type hint for `self` is equivalent to giving `Any` as a type hint.